Diagnostic/diagnostic_ut.py

```diff
@@ -0,0 +1,74 @@
+import unittest
```

Comment:
Rather than have the unit tests and the test driver living in the same directory as the source, seems it would be better to create a "tests" directory and put them there. As LAD is improved and changed over the coming months, more tests will be written; better to have a clean home for them.